clients/include/utility.hpp

```diff
@@ -527,6 +527,32 @@ void print_matrix(T* CPU_result, T* GPU_result, int m, int n, int lda)
                    double(GPU_result[i + j * lda].imag()));
 }
 
+/*! \brief  Debugging purpose, print out sparse matrix, not valid in complex number  */
+template <typename T, std::enable_if_t<!is_complex<T>, int> = 0>
+void print_sparse_matrix(int m, int n, int* csrRowPtr, int* csrColInd, T* csrVal)
```

Comment:
There is a also cholmod_print_sparse() routine to print a sparse matrix, but perhaps in compressed sparse column format.   Just FYI.

Reply:
I attempted to use that method, but the only thing it was printing out were the matrix dimensions and a validity check. If there's a way to get it to print the matrix elements, I haven't found how.